src/main/java/de/medizininformatikinitiative/torch/rest/FhirController.java

```diff
@@ -209,6 +226,11 @@ private Mono<Void> processCrtdl(Crtdl crtdl, String jobId) {
                 .then();  // This returns Mono<Void> indicating completion
     }
 
+    public Mono<List<String>> fetchPatientList(Crtdl crtdl) {
+
+        return (useCql) ? fetchPatientListUsingCql(crtdl): fetchPatientListFromFlare(crtdl);
```

Comment:
We need an abstraction for a patientList Fetcher here. The FhirController should get the right abstraction and not use a boolean flag here

Reply:
extra issue